services/trainer/trainer/utterance.py

```diff
@@ -0,0 +1,37 @@
+from dataclasses import dataclass, fields
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+
+@dataclass
+class Utterance:
+    audio_file_name: str
+    transcript: str
+    start_ms: Optional[int]
+    stop_ms: Optional[int]
+    speaker_id: Optional[str]
+
+    def combine(self, utterance: "Utterance") -> "Utterance":
+        return Utterance(
+            audio_file_name=self.audio_file_name,
+            transcript=f"{self.transcript} {utterance.transcript}",
+            start_ms=self.start_ms,
```

Comment:
In the next revision, I have just removed utterances having timestamps at all. 

The plan is (in preprocessing in the cloud functions) to split the audio files/transcripts into utterances so that each utterance has its own audio file of which it spans the entire length.